getnacelle/nacelle-js
# feat(ENG-6845): return `cart`, `userErrors` and `errors` from all client methods

## Why are these changes introduced?

Addresses [ENG-6845](https://nacelle.atlassian.net/browse/ENG-6845)

> Return `cart`, `userErrors` and `errors` from all client actions

## What is this pull request doing?

- Creates a new return type called `CartResponse`
- Creates a utility function called `formatCartResponse` to remove duplicate code across actions, and simplify testing
- Creates new tests, and updates existing tests to work with new format

## How to Test

1. Check out the `ENG-6845/return-cart-errors-userErrors` branch.
2. `npm run bootstrap` from the monorepo root, as needed.
3. `cd packages/shopify-cart`
4. `npm run test:ci` - tests should pass with expected coverage

<img width="379" alt="Screen Shot 2022-08-29 at 12 48 09 PM" src="https://user-images.githubusercontent.com/24395931/187252495-f83f9fdf-b751-41cd-83ae-3fac83838355.png">
